src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/scatterPlot.js

```diff
@@ -1,171 +1,560 @@
+/**
+ * Helper function to load scripts.
+ *
+ * It runs only once, so it won't load the same script on this or any other visualization show/hide.
+ */
 function loadScript(url) {
-    var script = document.createElement("script");
+    let script = document.createElement("script");
     script.src = url;
 
     document.head.appendChild(script);
 }
 
-loadScript('https://d3js.org/d3.v4.min.js');
+/**
+ * Helper function to load styles.
+ *
+ * It runs only once, so it won't load the same style on this or any other visualization show/hide.
+ */
+function loadStyle(url) {
+    let style   = document.createElement("link");
+    style.href  = url;
+    style.rel   = "stylesheet";
+    style.media = "screen";
+    style.type  = "text/css";
+
+    document.head.appendChild(style);
+}
+
+const label_style        = "font-family: DejaVuSansMonoBook; font-size: 10px;";
+const x_axis_label_width = 30;
+const label_padding_x    = 7;
+const label_padding_y    = 2;
```

Comment:
above you have var `x_asis_label_...` which tells what kind of label this is. Are these vars applied to every label, bot x and y?